src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/OperationCorrelationTelemetryInitializer.cs

```diff
@@ -1,35 +1,28 @@
 ﻿namespace Microsoft.ApplicationInsights.AspNetCore.TelemetryInitializers
 {
     using System;
-    using System.Net.Http.Headers;
     using Microsoft.ApplicationInsights.AspNetCore.Common;
     using Microsoft.ApplicationInsights.AspNetCore.DiagnosticListeners;
     using Microsoft.ApplicationInsights.Channel;
     using Microsoft.ApplicationInsights.DataContracts;
-    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility;
     using Microsoft.AspNetCore.Http;
-    using Microsoft.Extensions.Primitives;
 
     /// <summary>
     /// A telemetry initializer that will set the correlation context for all telemetry items in web application.
     /// </summary>
     internal class OperationCorrelationTelemetryInitializer : TelemetryInitializerBase
     {
-        private ICorrelationIdLookupHelper correlationIdLookupHelper = null;
+        private readonly IApplicationIdProvider applicationIdProvider;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="OperationCorrelationTelemetryInitializer"/> class.
         /// </summary>
         /// <param name="httpContextAccessor">Accessor for retrieving the current HTTP context.</param>
-        /// <param name="correlationIdLookupHelper">A store for correlation ids that we don't have to query it everytime.</param>
-        public OperationCorrelationTelemetryInitializer(
-            IHttpContextAccessor httpContextAccessor, ICorrelationIdLookupHelper correlationIdLookupHelper) : base(httpContextAccessor)
+        /// <param name="applicationIdProvider">Provider for resolving application Id to be used by Correlation.</param>
+        public OperationCorrelationTelemetryInitializer(IHttpContextAccessor httpContextAccessor, IApplicationIdProvider applicationIdProvider) : base(httpContextAccessor)
         {
-            if (correlationIdLookupHelper == null)
-            {
-                throw new ArgumentNullException(nameof(correlationIdLookupHelper));
-            }
-            this.correlationIdLookupHelper = correlationIdLookupHelper;
+            this.applicationIdProvider = applicationIdProvider ?? throw new ArgumentNullException(nameof(applicationIdProvider));
```

Comment:
can we avoid this exception logic here and instead check if appidprovider is null in Initialize?